UC NetCode Fall 2024/Assets/App/Resource/Scripts/TntScript.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections;
+using System.Collections.Generic;
+using Unity.Netcode;
+using UnityEngine;
+
+namespace App.Resource.Script.Obj
+{
+    public class TntScript : NetworkBehaviour
+    {
+        public float _fuseTimer = 4f;
+        public NetworkObject ExplosionPrefab;
+
+        public override void OnNetworkSpawn()
+        {
+            base.OnNetworkSpawn();
+            if (IsServer)
+            {
+                StartCoroutine("TriggerFuse");
+            }
+        }
+
+        private IEnumerator TriggerFuse()
+        {
+            Debug.Log("Fuse List");
+            yield return new WaitForSeconds(_fuseTimer);
+            TriggerExplosionRpc();
+        }
+
+        [Rpc(target: SendTo.Server)]
+        public void TriggerExplosionRpc()
+        {
+            NetworkObject explosive = NetworkManager.Instantiate(ExplosionPrefab, transform.position, transform.rotation);
+
+            explosive.Spawn(true);
+            this.NetworkObject.Despawn();
+        }
+
+        private void OnCollisionEnter(Collision other)
+        {
+            if (other.gameObject.tag.Equals("Projectile") || other.gameObject.tag.Equals("Explosion"))
+            {
+                TriggerExplosionRpc();
+            }
+        }
```

Comment:
Right now the Projectile and explosion tag only matter to the TnT script, so its important to make sure those are matching accordingly.

now in your Bullet Prefab you have it set to a tag of Bullet. You would need to update this OncollisionEnter script if you were trying to get it to work with this current tag to `other.gameObject.tag.Equals("Bullet")`
![image](https://github.com/user-attachments/assets/cd93d8b8-bb3b-453f-875f-8cd95ce28270)